src/convertToObject.js

```diff
@@ -6,7 +6,29 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.length === 0) {
+    return {};
+  }
+
+  // debugger;
+
+  const getArrayFromString = sourceString.split(';');
+
+  const formateString = getArrayFromString
+    .filter((arrString) => arrString.trim().length > 0)
+    .map((arrString) => {
+      const [key, value] = arrString.split(':').map((part) => part.trim());
+
+      return {
+        [key]: value
+          .split(', ') // добавив тут пробіл і все спрацювало, ну і шляпа
+          .map((v) => v.trim())
+          .join(','),
+      };
+    });
```

Comment:
[CODE STYLE] - All chained methods called on the same level should have same indentation. This makes code much more readable. Additionally, [CODE KNOWLEDGE] - Don't need to make copy before each operation. Moreover, methods like `filter` or `map` don't mutate the original array.